### ISSUE TITLE
Incompatibility of "nullspace" and 

### ISSUE THREAD

**Issue by ekawecki** [firedrakeproject/firedrake#1147]

**Comment by ekawecki** [comment]
Hi guys, I'm trying to implement the Neumann-Poisson problem: -Delta u = f, in \Omega, du/dn = 0 on \partial\Omega, with an extra parameter that is meant to represent the compatibility condition for the neumann problem, i.e., \int_{\partial\Omega}du/dn = \int_\Omega f. I've tried implementing it as follows:
Con = FunctionSpace(m, "R", 0)
   W = FunctionSpace(m, "CG", deg)
   S = W * Con
   u, mu = TrialFunctions(S)
   v, l = TestFunctions(S)
   V_basis = VectorSpaceBasis(constant=True)
   nullspace = MixedVectorSpaceBasis(S, [V_basis, S[1]])
   x, y  = SpatialCoordinate(m)
   n = as_vector([x/sqrt(pow(x,2)+pow(y,2)),y/sqrt(pow(x,2)+pow(y,2))])

   rho = pow(x,2)+pow(y,2)
   utrue = (1.0/6.0)*pow(rho,3)-(1.0/2.0)*rho+5.0/24.0
   utrue = utrue-assemble(utrue*dx(m, degree = quad_deg))/assemble(1.0*dx(m, degree = quad_deg))
   du0 = x*pow(rho,2)-x
   du1 = y*pow(rho,2)-y
   d2udxx = (5.0*pow(x,2.0)+pow(y,2.0))*rho-1.0#1-pow(y,2)/pow(rho,3.0/2.0)
   d2udxy = 4.0*x*y*rho
   d2udyy = (pow(x,2.0)+5.0*pow(y,2.0))*rho-1.0
   a = (u.dx(0)*v.dx(0)+u.dx(1)*v.dx(1))*dx(m, degree = quad_deg)\
       +(h**(-deg))*(u.dx(0)*n[0]+u.dx(1)*n[1]-mu)*(v.dx(0)*n[0]+v.dx(1)*n[1]-l)*ds(m,degree = quad_deg)\
   #+(h**(-(deg+1)))*u*v*ds(m)
   f = -(d2udxx+d2udyy)
   print "f domain integral:", assemble(f*dx(m))
   L = f*v*dx(m, degree = quad_deg)#+(pow(h,-deg))*(du0*n[0]+du0*n[1])*(v.dx(0)*n[0]+v.dx(1)*n[1])*ds(m, degree = domaindeg)
   Uh = Function(S)
   solve(a == L, Uh, nullspace = nullspace, solver_parameters={'ksp_type': 'preonly','pc_type': 'lu', 'pc_factor_mat_solver_package': 'mumps', 'ksp_rtol': tol,'snes_rtol': tol, 'snes_atol': tol})

But I end up with the following error:

 File "pureneumanntrial.py", line 67, in <module>
   solve(a == L, Uh, nullspace = nullspace, solver_parameters={'ksp_type': 'preonly','pc_type': 'lu', 'pc_factor_mat_solver_package': 'mumps', 'ksp_rtol': tol,'snes_rtol': tol, 'snes_atol': tol})
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/solving.py", line 124, in solve
   _solve_varproblem(*args, **kwargs)
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/solving.py", line 153, in _solve_varproblem
   appctx=appctx)
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/variational_solver.py", line 286, in __init__
   super(LinearVariationalSolver, self).__init__(*args, **kwargs)
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/variational_solver.py", line 177, in __init__
   transpose=False, near=False)
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/solving_utils.py", line 352, in set_nullspace
   nullspace._apply(self._jac, transpose=transpose, near=near)
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/nullspace.py", line 327, in _apply
   self._apply_monolithic(matrix, transpose=transpose, near=near)
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/nullspace.py", line 285, in _apply_monolithic
   self._build_monolithic_basis()
 File "/Users/kawecki/firedrake-test/firedrake/src/firedrake/firedrake/nullspace.py", line 256, in _build_monolithic_basis
   v /= v.dat.norm
 File "/Users/kawecki/firedrake-test/firedrake/src/PyOP2/pyop2/base.py", line 2152, in norm
   return sqrt(self.inner(self))
 File "/Users/kawecki/firedrake-test/firedrake/src/PyOP2/pyop2/base.py", line 2491, in inner
   ret += s.inner(o)
AttributeError: 'Global' object has no attribute 'inner'


It seems that monolithic and spaces with "R" are incompatible.

**Comment by dham** [comment]
You are correct. The way that R space matrices are distributed in parallel is not compatible with building a monolithic matrix. See http://www.firedrakeproject.org/r-space.html for more information on how R space matrices work.

**Comment by wence-** [comment]
This is slightly different, the problem is arising when building the nullspace for the full mixed space
